deploy/tests/network.ts

```diff
@@ -77,76 +81,320 @@ describe('network', () => {
+            const { deployer, deployerV2, foundationMultisig, daoMultisig, liquidityProtection, legacyStakingRewards } =
+                await getNamedAccounts();
+
+            await expectRoleMembers(
+                bntGovernance as any as AccessControlEnumerable,
+                Roles.TokenGovernance.ROLE_SUPERVISOR,
+                [isMainnetFork() ? foundationMultisig : deployer]
+            );
+            await expectRoleMembers(
+                bntGovernance as any as AccessControlEnumerable,
+                Roles.TokenGovernance.ROLE_GOVERNOR,
+                isMainnet() ? [deployerV2] : [deployer]
+            );
+            await expectRoleMembers(
+                bntGovernance as any as AccessControlEnumerable,
+                Roles.TokenGovernance.ROLE_MINTER,
+                isMainnet()
+                    ? [standardRewards.address, bntPool.address, liquidityProtection, legacyStakingRewards]
+                    : [standardRewards.address, bntPool.address]
+            );
+
+            await expectRoleMembers(
+                vbntGovernance as any as AccessControlEnumerable,
+                Roles.TokenGovernance.ROLE_SUPERVISOR,
+                [isMainnetFork() ? foundationMultisig : deployer]
+            );
+            await expectRoleMembers(
+                vbntGovernance as any as AccessControlEnumerable,
+                Roles.TokenGovernance.ROLE_GOVERNOR,
+                isMainnet() ? [deployerV2] : [deployer]
+            );
+            await expectRoleMembers(
+                vbntGovernance as any as AccessControlEnumerable,
+                Roles.TokenGovernance.ROLE_MINTER,
+                isMainnet() ? [bntPool.address, liquidityProtection] : [bntPool.address]
+            );
+
+            await expectRoleMembers(masterVault, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig, network.address]);
+            await expectRoleMembers(masterVault, Roles.Vault.ROLE_ASSET_MANAGER, [
+                network.address,
+                poolCollection.address
+            ]);
+
+            await expectRoleMembers(externalProtectionVault, Roles.Upgradeable.ROLE_ADMIN, [
+                daoMultisig,
+                network.address
+            ]);
+            await expectRoleMembers(externalProtectionVault, Roles.Vault.ROLE_ASSET_MANAGER, [
+                network.address,
+                poolCollection.address
+            ]);
+
+            await expectRoleMembers(externalRewardsVault, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig]);
+            await expectRoleMembers(externalRewardsVault, Roles.Vault.ROLE_ASSET_MANAGER, [standardRewards.address]);
+
+            await expectRoleMembers(poolTokenFactory, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig]);
+
+            await expectRoleMembers(networkSettings, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig]);
+
+            await expectRoleMembers(bntPool, Roles.Upgradeable.ROLE_ADMIN, [daoMultisig, network.address]);
```

Comment:
why does the network need the admin role of the BNT pool?

Reply:
It needs it to grant/revoke roles when processing the addition/removal of pool collections

Reply:
oh right 👍